src/ansys/units/dimensions.py

```diff
@@ -1,142 +1,100 @@
-
-    def _units_to_dim(
-        self, units: str, power: float = None, dimensions: list = None
-    ) -> list:
-        """
-        Convert a unit string into a dimensions list.
-
-        Parameters
-        ----------
-        units : str
-            Unit string of quantity.
-        power : float
-            Power of unit string.
-
-        Returns
-        -------
-        list
-            Dimensions representation of unit string.
-        """
-        power = power or 1.0
-        dimensions = dimensions or [0.0] * self.max_dim_len()
-
-        # Split unit string into terms and parse data associated with individual terms
-        for term in units.split(" "):
-            _, unit_term, unit_term_power = filter_unit_term(term)
-
-            unit_term_power *= power
-
-            # retrieve data associated with fundamental unit
-            if unit_term in ansunits._fundamental_units:
-                idx = (
-                    ansunits._dimension_order[
-                        ansunits._fundamental_units[unit_term]["type"]
-                    ]
-                    - 1
-                )
-                dimensions[idx] += unit_term_power
-
-            # Retrieve derived unit composition unit string and factor.
-            if unit_term in ansunits._derived_units:
-                # Recursively parse composition unit string
-                dimensions = self._units_to_dim(
-                    units=ansunits._derived_units[unit_term]["composition"],
-                    power=unit_term_power,
-                    dimensions=dimensions,
-                )
-
-        return dimensions
+    def __init__(self, dimensions_container: (dict[str, float], list[float]) = None):
+        self._mass = 0.0
+        self._length = 0.0
+        self._time = 0.0
+        self._temperature = 0.0
+        self._temperature_difference = 0.0
+        self._angle = 0.0
+        self._chemical_amount = 0.0
+        self._light = 0.0
+        self._current = 0.0
+        self._solid_angle = 0.0
+        if dimensions_container:
+            if len(dimensions_container) > Dimensions.max_dim_len():
+                raise DimensionsError.EXCESSIVE_DIMENSIONS(len(dimensions_container))
+
+            if isinstance(dimensions_container, dict):
+                for dim in dimensions_container:
+                    if dim in ansunits._dimension_order:
+                        private_dim = f"_{dim.lower().replace(' ','_')}"
+                        setattr(self, private_dim, dimensions_container[dim])
+                    else:
+                        raise DimensionsError.INCORRECT_DIMENSIONS()
+            else:
+                for idx, attr in enumerate(self.__dict__):
+                    if idx < len(dimensions_container):
+                        setattr(self, attr, dimensions_container[idx])
 
     @property
-    def units(self):
-        """Unit string representation of dimensions."""
-        return self._unit
+    def short_dictionary(self):
```

Comment:
I'm kind of wondering about the names here.  Short means non-zeros so:
```
dim = Dimensions(...)
dim.as_dict()
dim.as_dict(nonzeros=True) # default for nonzeros is False
dim.as_list(nonzeros=True)  # default for nonzeros is False
```
I'm not sure a list makes sense with `nonzeros=True`.  What do I do with that returned container if it does not tell me _which_ base quantities are non-zero.

Reply:
Do you mean something like this:
```
def as_dict(self, non_zero = False):
        dictionary = self.__dict__
        if non_zero:
            dictionary = {x:y for x,y in dictionary.items() if y!=0}
        return dictionary
```

Reply:
Yes, that is what I meant.